File: windows/db_explorer/modules/Logger.cs
namespace db_explorer.modules
{
    public class Logger : ILogger
    {
        public void Log(string level, string message)
        {
            System.Diagnostics.Debug.WriteLine($"[{level}] {message}");
            PlatformDriverManager.Current?.Emit(new DriverManagerOutboundMessage
            {
                Type = DriverManagerOutboundMessageType.Log,
                Data = new
                {
                    level,
                    message
                },
            });;
        }

        public void Debug(string message)
        {
            Log("Debug", message);
        }

        public void Info(string message)
        {
            Log("Info", message);
        }

        public void Warn(string message)
        {
            Log("Warn", message);
        }

        public void Error(string message)
        {
            Log("Error", message);
        }

        public void Fatal(string message)
        {
            Log("Fatal", message);
        }
    }
}

File: ios/db_explorer/modules/Logger.swift
class Logger {
  func Log(_ level: String, _ message: String) {
    let outboundMessage = DriverManagerOutboundMessage.Log(DriverManagerLogData(level: level, message: message))
    PlatformDriverManager.Current?.emit(outboundMessage)
  }
  
  func Info(_ message: String) {
    Log("Info", message)
  }
  
  func Warn(_ message: String) {
    Log("Warn", message)
  }
  
  func Debug(_ message: String) {
    Log("Debug", message)
  }
  
  func Error(_ message: String) {
    Log("Error", message)
  }
  
  func Fatal(_ message: String) {
    Log("Fatal", message)
  }
}
